otx/algorithms/common/tasks/base_task.py

```diff
@@ -96,6 +96,7 @@ class OTXTask(IInferenceTask, IExportTask, IEvaluationTask, IUnload, ABC):
     def __init__(self, task_environment: TaskEnvironment, output_path: Optional[str] = None):
         self._config: Dict[Any, Any] = {}
         self._task_environment = task_environment
+        self._model_name = task_environment.model_template.name
```

Comment:
Could you explain why this parameter should be added to base? I want we should carefully select new attribute or function to base. It is too complex before.

Reply:
It is required by both `SegmentationNNCFTask` and `MMSegmenationTask`. Before refactoring, it was attribute of `BaseTask`. Even though another task does not use `self._model_name`, I believe this location is appropriate to set `self._model_name`. If you have more good idea, please let me know.

Reply:
If it is only use for segmentaion task, then it should be goes to MMSegmentationTask

Reply:
Also, I'm curious why only segmentation use that attribute? Can you guess why?

Reply:
Not exactly for `SegmentationNNCFTask`, `MMSegmentationTask` requires to generate learning curves. (possibly due to Geti?)

Reply:
generating learning curve function are defined in OTXSegmentationTask. Therefore it should be moved to OTXSegmentationTask

Reply:
And I prefer just using task_environment.model_template.name in generating learning curve function if it is only required model name. But It's personal preference

Reply:
Could you review again? thanks!